src/app/collection/settings/tab/link-types/collection-link-types.component.ts

```diff
@@ -17,176 +17,108 @@
-import {selectAllCollections} from '../../../../core/store/collections/collections.state';
+import {I18n} from '@ngx-translate/i18n-polyfill';
+import {Observable, combineLatest as observableCombineLatest} from 'rxjs';
+import {LinkTypeModel} from '../../../../core/store/link-types/link-type.model';
+import {selectCollectionByWorkspace, selectCollectionsDictionary} from '../../../../core/store/collections/collections.state';
+import {filter, map, mergeMap, tap} from 'rxjs/operators';
+import {selectLinkTypesByCollectionId} from '../../../../core/store/link-types/link-types.state';
+import {CollectionModel} from '../../../../core/store/collections/collection.model';
+import {LinkTypesAction} from '../../../../core/store/link-types/link-types.action';
+import {LinkInstancesAction} from '../../../../core/store/link-instances/link-instances.action';
+import {Subscription} from 'rxjs/index';
+import {isNullOrUndefined} from 'util';
 
 @Component({
-  selector: 'collection-link-types',
   templateUrl: './collection-link-types.component.html',
   styleUrls: ['./collection-link-types.component.scss']
 })
-export class CollectionLinkTypesComponent extends CollectionTabComponent implements OnInit {
-
-  @ViewChildren('name')
-  public linkTypeNameInput: QueryList<ElementRef>;
+export class CollectionLinkTypesComponent implements OnInit, OnDestroy {
 
   public linkTypes: LinkTypeModel[] = [];
+  public collection: CollectionModel;
+  public searchString: String;
 
-  public collections: { [collectionId: string]: Collection } = {};
+  private subscriptions = new Subscription();
 
-  constructor(private linkTypeService: LinkTypeService,
-              private router: Router,
+  constructor(private i18n: I18n,
               private notificationService: NotificationService,
-              store: Store<AppState>) {
-    super(store);
+              private store: Store<AppState>) {
   }
 
-  public ngOnInit(): void {
-    super.ngOnInit();
-
-    this.store.select(selectAllCollections).subscribe(
-      collections => this.collections = collections.reduce((acc, coll) => {
-        acc[coll.id] = coll;
-        return acc;
-      }, {})
-    );
+  public ngOnInit() {
+    this.subscribeData();
   }
 
-  private initializeLinkTypes(): void {
-    this.linkTypes = [];
-    this.linkTypeService.getLinkTypes({collectionIds: [this.collection.id]}).pipe(
-      map(linkTypes => linkTypes.filter(linkType => linkType.collectionIds[0] === this.collection.id))
-    ).subscribe(
-      linkTypes => this.linkTypes = linkTypes.map(linkType => new LinkTypeModel(linkType)),
-      error => this.notificationService.error('Failed fetching LinkTypes')
+  private subscribeData() {
+    this.subscriptions.add(this.store.select(selectCollectionByWorkspace).pipe(
+      filter(collection => !!collection),
+      mergeMap(collection => this.selectLinkTypesForCollection(collection.id)))
+      .subscribe(linkTypes => this.linkTypes = linkTypes)
```

Comment:
What about assigning link types to `linkTypes$` and collections to `collection$` instead of subscribing to their changes? I can see they are both used only in the template.

If you also manage to make `BehaviourSubject` from `searchString$`, you can use `OnPush` change detection in this component since it seems to be well designed.

Reply:
Good idea :) Just a moment!